src/content/docs/fr/guides/cms/hashnode.mdx

```diff
@@ -0,0 +1,352 @@
+    };
+
+    ```
+
+3. Configurez `schema.ts` pour définir la forme des données renvoyées par l'API Hashnode.
+
+    ```ts title="src/lib/schema.ts"
+
+    import { z } from "astro/zod";
+
+    export const PostSchema = z.object({
+        author: z.object({
+            name: z.string(),
+            profilePicture: z.string(),
+            }),
+        publishedAt: z.string(),
+        title: z.string(),
+        subtitle: z.string(),
+        brief: z.string(),
+        slug: z.string(),
+        readTimeInMinutes: z.number(),
+        content: z.object({
+            html: z.string(),
+        }),
+        tags: z.array(z.object({
+            name: z.string(),
+            slug: z.string(),
+        })),
+        coverImage: z.object({
+            url: z.string(),
+        }),
+    })
+
+    export const AllPostsDataSchema = z.object({
+        publication: z.object({
+            title: z.string(),
+            posts: z.object({
+                pageInfo: z.object({
+                    hasNextPage: z.boolean(),
+                    endCursor: z.string(),
+                }),
+                edges: z.array(z.object({
+                    node: PostSchema,
+                })),
+            }),
+        }),
+    })
+
+    export const PostDataSchema = z.object({
+        publication: z.object({
+            title: z.string(),
+            post: PostSchema,
+        }),
+    })
+
+    export type Post = z.infer<typeof PostSchema>
+    export type AllPostsData = z.infer<typeof AllPostsDataSchema>
+    export type PostData = z.infer<typeof PostDataSchema>
+
+    ```
+</Steps>
+
+### Affichage d'une liste d'articles
+
+La recherche via `getAllPosts()` renvoie un tableau d'objets contenant les propriétés de chaque article telles que :
+- `title` - le titre de l'article
+- `brief` - l'affichage HTML du contenu de l'article
+- `coverImage.url` - l'URL source de l'image vedette de l'article
+- `slug` - le nom de l'article.
+
+Utilisez le tableau `posts` renvoyé par la recherche pour afficher une liste d'articles de blog sur la page.
+
+
+```astro title="src/pages/index.astro"
+---
+import { getAllPosts } from '../lib/client';
+
+const data = await getAllPosts();
+const allPosts = data.publication.posts.edges;
+
+---
+
+<html lang="en">
+    <head>
+        <title>Astro + Hashnode</title>
+    </head>
+    <body>
+
+        {
+            allPosts.map((post) => (
+                <div>
+                    <h2>{post.node.title}</h2>
+                    <p>{post.node.brief}</p>
+                    <img src={post.node.coverImage.url} alt={post.node.title} />
+                    <a href={`/post/${post.node.slug}`}>Lire plus</a>
+                </div>
+            ))
+        }
+    </body>
+</html>
+```
+
+### Générer des pages
+
+<Steps>
+1. Créez la page `src/pages/post/[slug].astro` pour [générer dynamiquement une page](/fr/guides/routing/#routes-dynamiques) pour chaque post.
+
+    <FileTree title="Project Structure">
```

Comment:
Same as above: the title doesn't seem to be used (or I looked wrong) but in case of change it's best to translate it.
```suggestion
    <FileTree title="Structure du projet">
```